File: trabalho-final-poo/Mapa.cpp
#include "Mapa.h"

Mapa::Mapa()
{
}

Mapa::~Mapa()
{
}

string Mapa::getEscala(){
	return escala;
}
string Mapa::getLocal(){
	return local;
}

void Mapa::setEscala (string escala){
	this->escala= escala;
}
void Mapa::setLocal (string local){
	this->local = local;
}
Mapa* Mapa::cadastra(){
	Mapa *m = new Mapa();
	string nome;
	string escala;
	string local;
	string tags;
	int qtd;
	cin.ignore();
	cout<<"Insira um nome:"<<endl;
	getline (cin, nome);
	m->setNome(nome);
	cout<<"Insira um escala:"<<endl;
	getline (cin,escala);
	m->setEscala(escala);
	cout<<"Insira um local:"<<endl;
	getline (cin,local);
	m->setLocal(local);
	cout<<"Insira a quantidade:"<<endl;
	cin >>qtd;
	m->setQtd(qtd);
	cin.ignore();
	while (tags != "sair"){
		cout<<"Insira uma tag: (ou escreva sair para encerrar)"<<endl;
		getline (cin, tags);
		cin.clear();
		fflush(stdin);
		if (tags !="sair"){
			m->addTag(tags);
		}
	}
	return m;
}

void Mapa::imprimeFichaTecnica(){
	int larguraCaixa = 70;
	cout  << endl << setfill('=')<< setw(larguraCaixa)<< "" << setfill(' ')<<endl;
	cout << "| Mapa " << setw(larguraCaixa - 7) << " |" << endl;
	Item::imprimeFichaTecnica();
	cout << "| Local: " << setw(larguraCaixa - 11) << this->local  << " |" << endl;
	cout << "| Escala: " << setw(larguraCaixa - 12) << this->escala<< " |" << endl;
	cout << setfill('=')<< setw(larguraCaixa)<< "" << setfill(' ')<<endl;
}
#include "Mapa.h"

File: acervoBiblioteca/src/main/java/com/poo2/acervobiblioteca/itens/Mapa.java
package com.poo2.acervobiblioteca.itens;

import java.util.Scanner;

public class Mapa extends Item{
    private String escala;
    private String local;

    public String getEscala() {
        return escala;
    }

    public void setEscala(String escala) {
        this.escala = escala;
    }

    public String getLocal() {
        return local;
    }

    public void setLocal(String local) {
        this.local = local;
    }

    @Override
    public void cadastra() {
        super.cadastra();
        Scanner scanner =new Scanner(System.in);
        System.out.println("Insira o local:");
        this.local= scanner.nextLine();
        System.out.println("Insira a escala:");
        this.escala = scanner.nextLine();
        cadastraTags();
    }

    @Override
    public void imprimeFichaTecnica() {
        System.out.println("Mapa:");
        super.imprimeFichaTecnica();
        System.out.printf("Local: %s\nEscala: %s\n", this.local, this.escala);
        System.out.println("============================================================================");
    }
}

